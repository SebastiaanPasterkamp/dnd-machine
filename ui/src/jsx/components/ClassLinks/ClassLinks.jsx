import React from 'react';

import BasicLinks from '../../hocs/BasicLinks';

export const ClassLinks = BasicLinks(['data', 'class'], ['admin']);

export default ClassLinks;
