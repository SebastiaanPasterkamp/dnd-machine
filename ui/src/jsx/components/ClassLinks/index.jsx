export { default, ClassLinks } from './ClassLinks';
