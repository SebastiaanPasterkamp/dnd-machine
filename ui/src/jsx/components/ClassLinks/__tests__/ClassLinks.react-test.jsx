import React from 'react';
import renderer from 'react-test-renderer';
import MockRouter from 'react-mock-router';

import {
    mockedApi,
} from '../../../../../tests/__mocks__';

import ClassLinks from '..';

describe('ClassLinks', () => {
    const props = {
        id: 1,
        currentUser: {
            role: ['admin'],
        },
        altStyle: true,
        children: <span>Hi</span>,
    };

    beforeEach(() => {
        fetch.mockImplementation( mockedApi({
            current_user: {},
        }) );
    });

    afterEach(() => fetch.resetMocks());

    it('should render with minimum props', () => {
        const tree = renderer.create(
            <MockRouter>
                <ClassLinks />
            </MockRouter>
        );

        expect(tree).toMatchSnapshot();
    });

    it('should render with full props', () => {
        const tree = renderer.create(
            <MockRouter>
                <ClassLinks
                    {...props}
                />
            </MockRouter>
        );

        expect(tree).toMatchSnapshot();
    });

    it('should render without current full props', () => {
        const tree = renderer.create(
            <MockRouter>
                <ClassLinks
                    {...props}
                    id={undefined}
                />
            </MockRouter>
        );

        expect(tree).toMatchSnapshot();
    });

    it('should not render links for non-admin', () => {
        const tree = renderer.create(
            <MockRouter>
                <ClassLinks
                    {...props}
                    currentUser={{
                        role: ['dm'],
                    }}
                />
            </MockRouter>
        );

        expect(tree).toMatchSnapshot();
    });
});
