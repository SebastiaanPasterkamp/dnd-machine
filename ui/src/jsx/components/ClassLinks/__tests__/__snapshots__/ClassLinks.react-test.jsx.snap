// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassLinks should not render links for non-admin 1`] = `
<div
  className="base-link-group nice-btn-group"
>
  <span>
    Hi
  </span>
</div>
`;

exports[`ClassLinks should render with full props 1`] = `
<div
  className="base-link-group nice-btn-group"
>
  <a
    className="cursor-pointer nice-btn-alt icon fa-eye"
    href={undefined}
    onClick={[Function]}
  >
    View
  </a>
  <a
    className="cursor-pointer nice-btn-alt icon fa-pencil"
    href={undefined}
    onClick={[Function]}
  >
    Edit
  </a>
  <span>
    Hi
  </span>
</div>
`;

exports[`ClassLinks should render with minimum props 1`] = `null`;

exports[`ClassLinks should render without current full props 1`] = `
<div
  className="base-link-group nice-btn-group"
>
  <a
    className="cursor-pointer nice-btn-alt icon fa-plus"
    href={undefined}
    onClick={[Function]}
  >
    New
  </a>
  <span>
    Hi
  </span>
</div>
`;
